### PR TITLE
adding ModuleConfig struct to NodeModulesSpec

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
this struct will hold the desired configuration of the module on the node (image, to be loaded or not, modprobe configuration etc'), and will be passed as is to the worker Pod as an input

### CODE REVIEW

**Comment by ybettan** [comment]
Are we starting to work on the implementation of v2 before the [proposal](https://github.com/kubernetes-sigs/kernel-module-management/pull/463) was merged?

File: api/v1beta1/nodemodulestate_types.go [hunk 1]
```diff
@@ -20,16 +20,20 @@ import (
...
-	Namespace      string `json:"namespace"`
+type ModuleConfig struct {
 	ContainerImage string `json:"containerImage"`
 	//+optional
 	InTreeModuleToRemove string       `json:"inTreeModuleToRemove,omitempty"`
 	Modprobe             ModprobeSpec `json:"modprobe"`
 	KmodLoaded           bool         `json:"kmodLoaded"`
 }
 
+type ModuleNodeSpec struct {
```
Comment (ybettan) [inline api/v1beta1/nodemodulestate_types.go:31]:
```suggestion
type NodeModuleSpec struct {
```

File: api/v1beta1/nodemodulestate_types.go [hunk 2]
```diff
@@ -20,16 +20,20 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type ModuleNodeSpec struct {
-	Name           string `json:"name"`
-	Namespace      string `json:"namespace"`
+type ModuleConfig struct {
```
Comment (ybettan) [inline api/v1beta1/nodemodulestate_types.go:23]: Shouldn't this file be in `api/v2/...`?

**Comment by yevgeny-shnaidman** [comment]
> Are we starting to work on the implementation of v2 before the [proposal](https://github.com/kubernetes-sigs/kernel-module-management/pull/463) was merged?
> 
> I still have some un-aswered questions on the config file passed to the worker pods.

@qbarrand lets answer all the questions officially in the Enhanement

File: api/v1beta1/nodemodulestate_types.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply api/v1beta1/nodemodulestate_types.go:23]: i would prefer to leave it for now in v1. The official API to customer is not changing, so i am not sure if we want to move it to v2

File: api/v1beta1/nodemodulestate_types.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply api/v1beta1/nodemodulestate_types.go:31]: i will change it

**Pull request merged**